### ISSUE TITLE
I am starting out at Yolov8 and I need help. I have trained a custom Yolov8 and had resized my training images to 640x640 using Roboflow. I wan to know if YOLOV8 resizes the images to the required input size on its own when training or do I have to manually resize them. And if I have to manually resize them can some one guide me how to do so? My image sizes are 1920x1080. Thank you

### ISSUE THREAD

**Issue by mkrushna12** [ultralytics/ultralytics#3344]
I am starting out at Yolov8 and I need help. I have trained a custom Yolov8 and had resized my training images to 640x640 using Roboflow. I wan to know if YOLOV8 resizes the images to the required input size on its own when training or do I have to manually resize them. And if I have to manually resize them can some one guide me how to do so? My image sizes are 1920x1080. Thank you

_Originally posted by @khanonenet in https://github.com/ultralytics/ultralytics/discussions/2303_

**Comment by aakash-chakraborty1995** [comment]
During training you can resize them automatically. 

# Train the model
model.train(data='coco128.yaml', epochs=100, imgsz=640)

**Comment by glenn-jocher** [comment]
@mkrushna12 during training in YOLOv8, you can resize your training images to the required input size automatically by specifying the desired image size using the `--imgsz` argument. In the example command you provided, the `imgsz=640` argument indicates that the images will be resized to 640x640 pixels during training.

By setting the `imgsz` argument to the desired size, YOLOv8 will handle the resizing of the images for you automatically during the training process.

This ensures that all images are consistently resized to the specified dimensions, allowing the model to learn from images of the same size throughout the training process.

I hope this explanation helps! If you have any further questions or need assistance with anything else, feel free to ask.

**Comment by pauline-viguier** [comment]
hi there 
To train YOLO on SAR images for detecting boats, do I need to train YOLO on small images or on images with an identical size of Sentinel?

Thanks

**Comment by glenn-jocher** [comment]
@pauline-viguier hi there! 👋

Training YOLO on SAR images to detect boats can be effectively done without needing to match the exact size of Sentinel images. YOLO models, including YOLOv8, can resize your training images to a specified input size during training, ensuring consistency. It's common to use sizes like `640x640` for training for a good balance between speed and accuracy, but you're free to experiment with sizes that suit your dataset and performance needs.

Here’s a snippet showing how to specify the image size during training:

```python
from ultralytics import YOLO

# Load a model, specify desired training image size (e.g., 640x640)
model = YOLO('yolov8n.yaml')

# Train the model
results = model.train(data='your_dataset.yaml', epochs=100, imgsz=640)
```

Keep in mind, the key is to prepare your dataset well and choose an image size that aligns with your computational resources and accuracy requirements. Happy training and good luck with detecting those boats! 🚤
